server.go

```diff
@@ -227,6 +228,12 @@ func (server *Server) handle(s *streamWrap) error {
 		return newServerError(err)
 	}
 
+	if server.authorize != nil && !server.authorize(s.stream.Conn().RemotePeer(), svcID.Name, svcID.Method) {
+		errMsg := fmt.Sprintf("client does not have permissions to this method, service name: %s, method name: %s", svcID.Name, svcID.Method)
+		resp := &Response{svcID, errMsg, serverErr}
```

Comment:
actually, `return newServerError(errMsg)` is the right way, just like all the other errors in this methods. I don't think this can work right now (it trigger writes to a closed stream at least).

Reply:
> it trigger writes to a closed stream at least

Not sure what you meant by this. Do you mean the use of `s.stream.Conn().RemotePeer()`?

Reply:
nevermind, just return the error. `sendResponse` is called when this function returns an error.